### ISSUE TITLE
LOW BUG - 17.12 - Select without choices in Editor causes no data display no matter is fixed in editor

### ISSUE THREAD

**Issue by atl2017** [smap-consulting/smapserver#167]
### Actual Behaviour:
When you use editor for a new form with select_one or multiple and you forget to include choices and try to save file, file is saved while show errors. Situation is that not matter if you fix and include choices, table becomes corrupted:

![image](https://user-images.githubusercontent.com/29519901/35011798-6c525db6-fad5-11e7-89c0-b9aa5606797f.png)

### Steps to reproduce the problem

I create a new form and save without include choices.  Choices are from external csv.

![selectwochoices1jpg](https://user-images.githubusercontent.com/29519901/35011808-779fedfa-fad5-11e7-96e3-d97502b4f666.JPG)

After notice error I include choices (fields from csv..)

![image](https://user-images.githubusercontent.com/29519901/35011894-c5308188-fad5-11e7-9171-4d2f9aeec1c5.png)

After input one record, nothing is displayed

(Optional) If this is reproducible on a Smap hosted server then include

**Comment by nap2000** [comment]
Interesting.  This would only affect select multiples.  If you had more than 100 choices for that select multiple in your csv file then this is going to happen due to the "performance" that has been added to restrict the number of external columns added to a table via CSV files.   The remedy to that would be to use "compressed" on the select multiple.

How big was the CSV file?

**Comment by atl2017** [comment]
Neil, in fact is your file choicesfile.csv. It has few records.

![image](https://user-images.githubusercontent.com/29519901/35019949-b0388c2c-faf7-11e7-9016-3aae3e6b3c91.png)

**Comment by atl2017** [comment]
It is fixed if you reload xlsform and strated again. I do not close issue because the potential issue should be fixed, but is low priority.

![image](https://user-images.githubusercontent.com/29519901/35281175-4be0ddd2-0020-11e8-8e2b-0884befb8da2.png)

**Comment by atl2017** [comment]
Checked again and works fine with FT but web forms does not show option to select no matter I reload csv file again. My recommendation consider "No Choices" as an error and avoid saving.

I tested the same csv include both select one and select multiple question with choices included. 

![image](https://user-images.githubusercontent.com/29519901/39427316-f398ecee-4c48-11e8-96f0-6c92c525f596.png)

**Comment by nap2000** [comment]
I tried the following:

1. Select create form to go to online editir
2. Add a select one question
3. Set appearance to search('choices')
4. Select Save
5. Select Test 
No options show in options list

6. Back in editor select choices for the question
7. Add a name of "value" and a label of "label2"
8. Select Save
9. Select Test
10. Refresh browser

Options are available in options list

So is this fixed now?

**Comment by atl2017** [comment]
Tested Ok. 

I saw the "warning" icon with "1" but I could not find how to see message at all. In order to be fail-safe it could be good to find an easier way (or an alternative way) to see this warnings.

![image](https://user-images.githubusercontent.com/29519901/41093205-54ed6d3c-6a10-11e8-87c1-5db9b9f27bdb.png)

![image](https://user-images.githubusercontent.com/29519901/41093300-92583922-6a10-11e8-8a76-c44ef874bf4e.png)
